privacyidea.py

```diff
@@ -119,47 +150,47 @@ def authenticate(self, configurationAttributes, requestParameters, step):
             fm.add(FacesMessage.SEVERITY_ERROR, "Failed to communicate to privacyIDEA. Possible misconfiguration. Please have the administrator check the log files.")
             return False
 
-        authenticationService = CdiUtil.bean(AuthenticationService)
         identity = CdiUtil.bean(Identity)
 
         if step == 1:
             credentials = identity.getCredentials()
-            user_name = credentials.getUsername()
-            user_password = credentials.getPassword()
-
-            if user_name:
-                self.addToSession("currentUser", user_name)
-
-                response = self.pi.validateCheck(user_name, user_password)
-                if response:
-                    # First check if what was entered is sufficient to log in
-                    if response.getValue():
-                        logged_in = authenticationService.authenticate(user_name)
-                        if logged_in:
-                            self.addToSession("auth_success", True)
-                        return logged_in
-
-                    # If not, check if transaction was triggered
+            username = credentials.getUsername()
+            password = credentials.getPassword()
+
+            if username:
+                if self.sendPassword:
+                    response = self.pi.validateCheck(username, password)
+                    if response:
+                        # First check if what was entered is sufficient to log in
+                        if response.getValue():
+                            logged_in = self.login(username, password)
+                            if logged_in:
+                                self.addToSession("auth_success", True)
+                            return logged_in
```

Comment:
You return logged_in without a user "added" to the "Session".

Reply:
in this one case (step 1 and privacyidea returned value=true) we do not need to store the user+password because there won't be a second step. The addToSession is just for our script to retain things for the second step, the Gluu server identifies the user by ``authenticationService.authenticate(username, password)`` which is done in ``self.login``.

Reply:
OK I understand. No 2nd step. No need to put things into the session.. thx